packages/dev/core/src/Physics/v2/physicsBody.ts

```diff
@@ -368,31 +368,16 @@ export class PhysicsBody {
     }
 
     /**
-     * Register a collision callback that is called when the body collides
-     * Filtering by body is inefficient. It's more preferable to register a collision callback for the entire world
-     * and do the filtering on the user side.
+     * Returns an observable that will be notified for all collisions happening for event-enabled bodies
+     * @returns Observable
      */
-    public registerOnCollide(
-        func: (collider: PhysicsBody, collidedAgainst: PhysicsBody, point: Nullable<Vector3>, distance: number, impulse: number, normal: Nullable<Vector3>) => void
-    ): void {
-        return this._physicsPlugin.registerOnBodyCollide(this, func);
-    }
-
-    /**
-     * Unregister a collision callback that is called when the body collides
-     */
-    public unregisterOnCollide(
-        func: (collider: PhysicsBody, collidedAgainst: PhysicsBody, point: Nullable<Vector3>, distance: number, impulse: number, normal: Nullable<Vector3>) => void
-    ): void {
-        return this._physicsPlugin.unregisterOnBodyCollide(this, func);
+    public getCollisionObservable(): Observable<CollisionEvent> {
+        return this._physicsPlugin.getCollisionObservable(this);

```

Comment:
Another question here - wouldn't this observable notify for each collision, even unrelated to this specific body? why don't we have collision per body?

Reply:
There is collision per body :slower and more memory intensive but easier to use. And global collision CB : way more efficient but user is in charge of filtering. it plugin specific implementation.

Reply:
Do you mean on the plugin level? because on the physics body level it seems like the 2nd approach is the only one available..
Maybe we can filter on our own using the observable's mask field? i.e. - have a unique binary-based ID for each body added and pass this ID as a mask when notifying the observers? I'm just thinking about all of the callbacks that will have a first "if" step